dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6561,6 +6562,42 @@ bool isForeignTableUpdate(THD* thd)
     return false;
 }
 
+bool isMCSTableUpdate(THD* thd)
+{
+    LEX* lex = thd->lex;
+
+    if ((lex->sql_command != SQLCOM_UPDATE_MULTI) &&
+        (lex->sql_command != SQLCOM_UPDATE))
+        return false;
+
+    Item_field* item;
+    List_iterator_fast<Item> field_it(lex->first_select_lex()->item_list);
```

Comment:
It is worth to at least put an assert before this ctor, namely assert(lex->first_select_lex()->item_list);

Reply:
`lex->first_select_lex()->item_list` is a `List<Item>` type. I don't think putting an assert on this makes much sense.